src/components/application_manager/src/help_promt_manager.cc

```diff
@@ -0,0 +1,418 @@
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "application_manager/message_helper.h"
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/commands/command_impl.h"
+#include "utils/logger.h"
+#include "utils/timer_task_impl.h"
+#include "utils/make_shared.h"
+
+#define TIMER_IDENTIFICATION \
+  "HelpPromtManager timer helps to buffer the requests"
+#define LIMIT_COMMAND 30
+#define TIMEOUT 10000
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "HelpPromtManager")
+
+namespace application_manager {
+
+HelpPromtManager::HelpPromtManager(Application* app)
+    : internal_timer(TIMER_IDENTIFICATION,
+                     new ::timer::TimerTaskImpl<HelpPromtManager>(
+                         this, &HelpPromtManager::onTimerExpired))
+    , commands_lock_(true)
+    , app_(*app)
+    , app_manager_(nullptr)
+    , type_sending_(NoneSend)
+    , count_timer_commands(0)
+    , count_requests_commands(0)
+    , is_tts_send_(false)
+    , is_ui_send_(false) {
+  requests_[SendHelpPromt] = &HelpPromtManager::SendTTSRequest;
+  requests_[SendVRHelp] = &HelpPromtManager::SendUIRequest;
+  requests_[SendAll] = &HelpPromtManager::SendAllRequests;
+}
+
+HelpPromtManager::~HelpPromtManager() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  onTimerStop();
+}
+
+void HelpPromtManager::OnVrCommandAdd(
+    const uint32_t& cmd_id, const smart_objects::SmartObject& command) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_DEBUG(logger_, "for appID:" << app_.app_id());
+  sync_primitives::AutoLock lock(commands_lock_);
+  if (NoneSend == type_sending_) {
+    return;
+  }
+  auto it = vr_commands_.find(cmd_id);
+  if (vr_commands_.end() == it) {
+    if (command.keyExists(strings::vr_commands)) {
+      const smart_objects::SmartObject& commands =
+          command[strings::vr_commands];
+      unsigned limit = LIMIT_COMMAND - count_requests_commands;
```

Comment:
@SNiukalov you don't this initialisation because you have if-else right after that which re-assigns this variable value.
Suggest:
```
int32_t commands_count_to_add = commands.length();
const int32_t commands_left = LIMIT_COMMAND - count_requests_commands;
if (commands_left <= 0) {
return; // no commands left
}
if (commands_left < commands_count_to_add) {
  commands_count_to_add = commands_left;
}